core/src/main/java/nl/nn/adapterframework/http/rest/ApiListenerServlet.java

```diff
@@ -212,6 +214,34 @@ protected void service(HttpServletRequest request, HttpServletResponse response)
 					log.warn(createAbortingMessage(remoteUser,405) + "method ["+method+"] not allowed");
 					return;
 				}
+				// set security handler
+				String authorizationHeader = request.getHeader("Authorization");
+				if(StringUtils.isNotEmpty(authorizationHeader) && authorizationHeader.contains("Bearer")) { // assumes that the jwt token is provided via authorization header
+					if(StringUtils.isEmpty(listener.getJwksURL())) {
+						response.sendError(401, "no jwksUrl supplied!");
+						return;
+					}
+					URL url = new URL(listener.getJwksURL());
+
+					JwtWrapper jwtWrapper = new JwtWrapper();
+					jwtWrapper.setToken(authorizationHeader.substring(7));
+					jwtWrapper.setJwksURL(url);
+					jwtWrapper.setRequiredIssuer(listener.getRequiredIssuer());
+					jwtWrapper.setRequiredClaims(listener.getRequiredClaims());
+					jwtWrapper.setExactMatchClaims(listener.getExactMatchClaims());
+
+					try {
+						JwtSecurityHandler securityHandler = new JwtSecurityHandler(request, jwtWrapper);
```

Comment:
The ApiListener should create a JwtValidator in open(), and reuse that for each request. This allows the JwtValidator to cache the public keys it retrieves from the jwksUrl.